metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/indexing_master.py

```diff
@@ -138,6 +142,30 @@ def status(self, env):
     def restart(self, env):
         from params import params
         env.set_params(params)
+        commands = IndexingCommands(params)
+        if params.ra_indexing_writer == 'Solr':
+            # Install Solr schemas
+            try:
+                if not commands.is_solr_schema_installed():
+                    commands.solr_schema_install(env)
+                    commands.set_solr_schema_installed()
+
+            except Exception as e:
```

Comment:
Can we move this try/catch block into `solr_schema_install`?  Then it doesn't have to be duplicated in `start` and `restart`.

Reply:
Done , Thanks @nickwallen